### ISSUE TITLE
Discrepancy between reported ViT-B-32 O-LVIS accuracy and reproduced accuracy

### ISSUE THREAD

**Issue by atharva-loci** [3dlg-hcvc/DuoduoCLIP#8]
Hello,

Thanks for the great paper and releasing the code!
I've been trying to reproduce the numbers shown in Fig4 of the paper where you show the effect of increasing the number of frames comparing the ViT-B-32, ViT-bigG-14 and DuoDuoCLIP models. 
The one frame top1 accuracy for B-32 is shown to be 28.23% on O-LVIS, when I run this model using the `transformers` package and use the "front" view render of the image I get a top1 accuracy of 42.7%. Additionally, I also see the performance drop when I increase the number of views and average the embedding: 2 views is 41.6%, 6 views is 40.3%.

I wanted to check how you calculate these baseline top1 accuracy numbers for the B-32 model. 
What strategy have you used when combining multiple frames? My tests run the different frames using batched inference and then average and normalise before computing the class similarity for the final prediction.
Which render from the original Zero123 dataset do you use for the one frame performance on the B-32, BigG-14 and DuoDuoCLIP results?

From your setup / knowledge, are there any other reasons why there would be such a big difference in performance?

Thanks!

**Comment by hanhung** [comment]
Hi,

Thank you for taking interest in our work. The method you use for aggregating the multi-view embeddings seems correct (averaging then normalizing). I was wondering:

1. How to you get embeddings for the 1156 classes? Are you using our class embeddings or calculating them yourself?
2. When you say front view do you mean at azimuth 0 and elevation 0 renders? For us Zero123 renders 12 random views for each object. During accuracy calculation we randomly sample M views (eg. 1) from it. It could be that the one front view had better vision of the object compared to the random view, but I don't expect it to differ so much.
3. We use the [OpenCLIP](https://github.com/mlfoundations/open_clip) implementation (```ViT-B-32, laion2b_s34b_b79k``` and ```ViT-bigG-14, laion2b_s39b_b160k```) not sure if there are differences with the ```transformers``` versions.

**Comment by atharva-loci** [comment]
Thanks for the reply.

1. I computed the embedding for the LVIS classes myself (This was without using templates like used in OpenShape and other papers)
2. The front view render used is not exactly (0, 0) azimuth and elevation. It uses (0.5, 0.9), the idea is to have a render that looks at the front face of the asset from a slightly elevated viewpoint. Using random renders from Zero123 could explain some of the difference but the absolute difference is quite large.
3. I used the same model in transformers, I will try to reproduce using the open_clip package.
